lib/vm.js

```diff
@@ -286,6 +287,36 @@ function runInThisContext(code, options) {
   return createScript(code, options).runInThisContext(options);
 }
 
+function compileFunction(code, params, options) {
+  const defaults = {
```

Comment:
Can you use default values in the de structuring instead of a defaults object?

Reply:
That seemed a better option, true, but now I'm also using `defaults` for error checking, so I guess it's good enough?